src/Common/src/TypeSystem/Ecma/EcmaType.cs

```diff
@@ -42,6 +43,19 @@ internal EcmaType(EcmaModule module, TypeDefinitionHandle handle)
 #endif
         }
 
+        public override int GetHashCode()
+        {
+            int nameHash = TypeHashingAlgorithms.ComputeNameHashCode(Name);
+            if (ContainingType == null)
```

Comment:
ContainingType is not super cheap - fetch it into a local variable, so that it does not need to be computed twice?